server/apps/immich/src/api-v1/album/album-repository.ts

```diff
@@ -208,6 +212,48 @@ export class AlbumRepository implements IAlbumRepository {
     return this.albumRepository.save(album);
   }
 
+  /**
+   * Makes sure all thumbnails for albums are updated by:
+   * - Removing thumbnails from albums without assets
+   * - Removing references of thumbnails to assets outside the album
+   * - Setting a thumbnail when none is set and the album contains assets
+   *
+   * @returns Amount of updated album thumbnails or undefined when unknown
+   */
+  async updateThumbnails(): Promise<number | undefined> {
+    // Subquery for getting a new thumbnail.
+    const newThumbnail = this.assetRepository
```

Comment:
Is there a way to not use queryBuilder for this? Just wondering.

Reply:
Highly doubt that's possible. The many-to-many relation between asset and album isn't explicitly defined, so there's no way to access it. Also for subqueries the query builder seems the only option.